sonic_db_config/db_config.go

```diff
@@ -18,223 +15,210 @@ const (
-	hostname, ok := inst["hostname"]
-	if !ok {
-		panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
+		err = DbInit()
+		if err != nil {
+			return "", err
+		}
 	}
-	return hostname.(string)
+	defer CatchException(&err)
+	hostname = swsscommon.SonicDBConfigGetDbHostname(db_name, ns)
+	return hostname, err
 }
 
-func GetDbPort(db_name string, ns string) int {
+func GetDbPort(db_name string, ns string) (port int, err error) {
 	if !sonic_db_init {
-		DbInit()
-	}
-	inst := GetDbInst(db_name, ns)
-	port, ok := inst["port"]
-	if !ok {
-		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+		err = DbInit()
+		if err != nil {
+			return -1, err
+		}
 	}
-	return int(port.(float64))
+	defer CatchException(&err)
+	port = swsscommon.SonicDBConfigGetDbPort(db_name, ns)
+	return port, err
 }
 
-func GetDbTcpAddr(db_name string, ns string) string {
+func GetDbTcpAddr(db_name string, ns string) (addr string, err error) {
 	if !sonic_db_init {
-		DbInit()
-	}
-	hostname := GetDbHostName(db_name, ns)
-	port := GetDbPort(db_name, ns)
-	return hostname + ":" + strconv.Itoa(port)
-}
-
-func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
-	data, err := io.ReadFile(SONIC_DB_GLOBAL_CONFIG_FILE)
-	if err == nil {
-		//Ref:https://stackoverflow.com/questions/18537257/how-to-get-the-directory-of-the-currently-running-file
-		dir, err := filepath.Abs(filepath.Dir(SONIC_DB_GLOBAL_CONFIG_FILE))
-		if err != nil {
-			panic(err)
-		}
-		sonic_db_global_config := make(map[string]interface{})
-		err = json.Unmarshal([]byte(data), &sonic_db_global_config)
+		err = DbInit()
 		if err != nil {
-			panic(err)
-		}
-		for _, entry := range sonic_db_global_config["INCLUDES"].([]interface{}) {
-			ns, ok := entry.(map[string]interface{})["namespace"]
-			if !ok {
-				ns = SONIC_DEFAULT_NAMESPACE
-			}
-			_, ok = ns_to_cfgfile_map[ns.(string)]
-			if ok {
-				panic(fmt.Errorf("Global Database config file is not valid(multiple include for same namespace!"))
-			}
-			//Ref:https://www.geeksforgeeks.org/filepath-join-function-in-golang-with-examples/
-			db_include_file := filepath.Join(dir, entry.(map[string]interface{})["include"].(string))
-			ns_to_cfgfile_map[ns.(string)] = db_include_file
-		}
-		if len(ns_to_cfgfile_map) > 1 {
-			sonic_db_multi_namespace = true
-		} else {
-			sonic_db_multi_namespace = false
-		}
-
-	} else if errors.Is(err, os.ErrNotExist) {
-		// Ref: https://stackoverflow.com/questions/23452157/how-do-i-check-for-specific-types-of-error-among-those-returned-by-ioutil-readfi
-		ns_to_cfgfile_map[SONIC_DEFAULT_NAMESPACE] = SONIC_DB_CONFIG_FILE
-		// Tests can override the file path via an env variable
-		if f, ok := os.LookupEnv("DB_CONFIG_PATH"); ok {
-			ns_to_cfgfile_map[SONIC_DEFAULT_NAMESPACE] = f
+			return "", err
 		}
-		sonic_db_multi_namespace = false
-	} else {
-		panic(err)
 	}
+	defer CatchException(&err)
+	hostname, err := GetDbHostName(db_name, ns)
```

Comment:
Should we keep using err instead of new err variable here GetDbHostName() and below GetDbPort()? since CatchException(&err) seems to be overwritten ambiguously?